components/playground/utils/utils.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"time"
+)
+
+// RetryOption is options for Retry()
+type RetryOption struct {
+	Attempts int64
+	Delay    time.Duration
+	Timeout  time.Duration
+}
+
+// default values for RetryOption
+var (
+	defaultAttempts int64 = 20
+	defaultDelay          = time.Millisecond * 500 // 500ms
+	defaultTimeout        = time.Second * 10       // 10s
+)
+
+// Retry retries the func until it returns no error or reaches attempts limit or
+// timed out, either one is earlier
+func Retry(doFunc func() error, opts ...RetryOption) error {
+	var cfg RetryOption
+	if len(opts) > 0 {
+		cfg = opts[0]
+	} else {
+		cfg = RetryOption{
+			Attempts: defaultAttempts,
+			Delay:    defaultDelay,
+			Timeout:  defaultTimeout,
+		}
+	}
+
+	// timeout must be greater than 0
+	if cfg.Timeout <= 0 {
```

Comment:
How about setting options automatically for invalid value like `cfg. Delay`?

Reply:
Just copied these lines from tiup-cluster